src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1514,23 +1514,62 @@ def __find_disk_from_topology(self, label, pool):
 
         return True
 
-    @accepts()
-    async def unlock_services_restart_choices(self):
+    @accepts(Int('id'))
+    async def unlock_services_restart_choices(self, oid):
         """
         Get a mapping of services identifiers and labels that can be restart
         on volume unlock.
         """
-        svcs = {
+        pool = await self._get_instance(oid)
+
+        services = {
             'afp': 'AFP',
             'cifs': 'SMB',
             'ftp': 'FTP',
             'iscsitarget': 'iSCSI',
             'nfs': 'NFS',
             'webdav': 'WebDAV',
-            'jails': 'Jails/Plugins',
-            'vms': 'Virtual Machines',
         }
-        return svcs
+
+        result = {}
+
+        for k, v in services.items():
+            service = await self.middleware.call('service.query', [['service', '=', k]], {'get': True})
+            if service['enable'] or service['state'] == 'RUNNING':
+                result[k] = v
+
+        try:
+            activated_pool = await self.middleware.call('jail.get_activated_pool')
+        except Exception:
+            activated_pool = None
+        # If iocage is not activated yet, there is a chance that this pool might have it activated there
```

Comment:
We could perhaps remove this comment though :)

Reply:
For me it looked like quite illogical thing, that's why I've put this comment, why do you want to remove it?

Reply:
Well that call will now never activate iocage nor if we start these jails with `rc_action`, so this comment isn't reflecting reality

Reply:
Hm, now, if I unlock pool that had iocage activated on it, it won't bring jails back?

Reply:
So when we import the pool, and if iocage wasn't already active and we execute `rc_action`, there are 2 cases:
1) Pool had jails, in which case the rc_action call should proceed as desired because the `iocage_set_up` function which checks if iocage is activated or not only checks if we have pool and dataset with iocage root dataset on it. So they should succeed.
2) Pool didn't had jails, the `iocage_set_up` call would return false and `rc_action` won't do anything with the pool remaining deactivated

Reply:
So, based on #1, after pool import iocage will become activated?

Reply:
Yes, that's how it should/will work. I can test this on my end to be certain as well - but if it's not, that's a bug and we have to fix it.

Reply:
So this comment does reflect reality? :)

Reply:
Also this change makes sense from another perspective, if a user was using iocage on an encrypted pool and then locked it. Now he is using it on a different pool, we shouldn't offer him jail restart option as it will conflict on which pool to use with both showing up as active ( some thing for me to look at to make sure we don't error out if someone manually does this )

Reply:
You're right 😅